File: Structural/Adapter/adapter.cs
public interface ITarget { string GetRequest(); }

class Adaptee {
    public string GetSpecificRequest() => "Specific request.";
}

class Adapter : ITarget {
    private readonly Adaptee _adaptee;
    public Adapter(Adaptee adaptee) => _adaptee = adaptee;
    public string GetRequest() => $"This is '{_adaptee.GetSpecificRequest()}'";
}

File: Structural/Adapter/adapter.py
class Adaptee:
    def specific_request(self) -> str:
        return "Specific request."

class Target:
    def request(self) -> str:
        return "Target: The default target's behavior."

class Adapter(Target):
    def __init__(self, adaptee: Adaptee):
        self.adaptee = adaptee
    def request(self) -> str:
        return f"Adapter: (TRANSLATED) {self.adaptee.specific_request()}"